src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1660,8 +1350,8 @@ public Boolean visitLink(LinkTree node, Content content) {
                         }
                         content.add(label);
                     } else {
-                        Content c = seeTagToContent(element, node, context.within(node));
-                        content.add(c);
+                        TagletWriterImpl t = getTagletWriterInstance(context.within(node));
```

Comment:
Separate to this PR: when I see a cast to or a variable declaration of the `TagletWriterImpl` type, I cannot help but wonder why most of the methods in `TagletWriter` are protected. Do you have any idea how we ended up like that?

Reply:
My reverse-engineered understanding is that using `protected` allows the methods to be overridden/implemented, but discourages direct calling, in favor of using `get[Block|Inline]TagOutput`, which is the way to get to those tag-specific methods. For better or worse, that does allow the build-in behavior to be modified.

The whole `TagletWriter` system is somewhat questionable, but revising it would be something of a Big Deal, and does not seem like the most important issue to deal with at this time.